### PR TITLE
tests: settings: Reducing duplicated code in FS tests

### PR DESCRIPTION
**Pull request by de-nordic**
Common code has been moved out of FS specific source into common
source files.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

### CODE REVIEW

**Comment by Laczen** [comment]
@de-nordic, nice cleanup thanks. Just a nitpick below.

File: tests/subsys/settings/fs/include/settings_test_fs.h [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+
+void config_empty_lookups(void);
+void test_config_insert(void);
+void test_config_getset_unknown(void);
+void test_config_getset_int(void);
+void test_config_getset_int64(void);
+void test_config_commit(void);
+
+void config_setup_nffs(void);
```
Comment (Laczen) [inline tests/subsys/settings/fs/include/settings_test_fs.h]: shouldn't this be inside the include settings_test.h in the nffs folder?

Comment (de-nordic) [reply tests/subsys/settings/fs/include/settings_test_fs.h]: This line shouldn't be here. settings_test_fs.h has been created from settings_test.h of nffs subdir and I have forgotten to remove this line.

**Review by Laczen — APPROVED** [review]

**Review by nvlsianpu — COMMENTED** [review]
Why not get rid of both tests/subsys/settings/littlefs/src/settings_test.h and tests/subsys/settings/littlefs/src/settings_test.h so new header "settings_test_fs.h" will be used?

**Comment by de-nordic** [comment]
> 
> 
> Why not get rid of both tests/subsys/settings/littlefs/src/settings_test.h and tests/subsys/settings/littlefs/src/settings_test.h so new header "settings_test_fs.h" will be used?

settings_test.h, which is different for littlefs and nffs provide FS specific definitions like mount point fine paths, etc.

**Review by nvlsianpu — APPROVED** [review]

**Comment by nvlsianpu** [comment]
@carlescufi @ioannisg This can be merged.

**Pull request merged**